### ISSUE TITLE
b64.js file missing

### ISSUE THREAD

**Issue by mankash** [strikeout/mylar#10]
Hello,

While trying to add the search functionality, following your readme (that's also the default readme on the github root page), you instruct us to: 

<===================================================
 "add the search package to the application meteor add search"
<===================================================

 On executing the above command, the following error is reported: 
<===================================================
 => Errors while adding packages:             

While building package search:
error: File not found: b64.js
<===================================================

The absence of b64.js can be further verified by: 
1> grep-ing through the code
2> searching for  b64.js via the github repository search feature in the site. 

Please add the missing file. 

Best!

**Comment by cedricvanrompay** [comment]
Same problem here

**Comment by cedricvanrompay** [comment]
OK, I may have found the beginning of the solution.

There is a file `base64.js` in the history of the repos:

https://github.com/strikeout/mylar/blob/313b60b27c58a3602c2049e566f955ec20d4ffb0/packages/base64/base64.js

I downloaded it to `packages/search` and renamed it `b64.js`,

Then `meteor add search` works.

However I cannot manage to search, when I click the "search in all rooms" button
my javascript console prints:

```
Error: principal does not exist in db
```

I don't know if this comes from the same issue or has another cause.

**Comment by cedricvanrompay** [comment]
I just noticed there is an issue, opened by a dev, titled _"Get the search-over-encrypted-documents working, test it"_ (issue #4).

So it seems search doesn't even work for the dev.

**Comment by ladieman217** [comment]
@cedricvanrompay same problem ... how to deal with it ? is there a way to make search work?

**Comment by cedricvanrompay** [comment]
What I did is using the original implementation from MIT:

    git clone -b public git://g.csail.mit.edu/mylar

**Comment by ladieman217** [comment]
@cedricvanrompay I am trying your method, by the way, do you use mylar in your production projects?any advise? thank you very much

**Comment by ladieman217** [comment]
I found this repo hasn't been updated for almost two years, I am afraid of the compatibility with the latest meteor release.

**Comment by cedricvanrompay** [comment]
No, I didn't use it in production and I would recommend not to. Yes it is probably not compatible with a lot of things; even with the (old) proper version of Mylar it crashes a lot.

I used mylar to write a Proof-of-Concept implementation showing that the crypto protocol they use for searching on encrypted data does not meet the security level they claim.
Paper to appear in PETS 2017 conference: https://petsymposium.org/2017/paperlist.php
(title "A Leakage-Abuse Attack Against Multi-User Searchable Encryption")

Proof-of-concept will be available here: http://www.eurecom.fr/~vanrompa/iterative-testing/

As a general remark, mylar was not built to be used in production but for research purpose, so using it in production was a bad idea to start with.

**Comment by ladieman217** [comment]
@cedricvanrompay thank you very much

**Comment by ladieman217** [comment]
@cedricvanrompay hi, I have been playing with mylar for a few weeks, now I am wondering can we use mylar with any other framework other than meteor? because normal web apps are built with REST api and front end framework such as Angular or React, any advice ? thank you very much..
